routing.go

```diff
@@ -51,7 +49,7 @@ type routelet struct {
 // lifetime of the routelet is bound to the provided context. When the context is
 // cancelled, the routelet stops tracking the latest routing information.
 func newRoutelet(ctx context.Context, env env, group string) *routelet {
-	r := &routelet{env: env, balancers: map[string]*routingBalancer{}}
+	r := &routelet{balancers: map[string]*routingBalancer{}}
 	go r.watchRoutingInfo(ctx, env, group)
```

Comment:
I've noticed there are a couple places where goroutines are started in the "constructor